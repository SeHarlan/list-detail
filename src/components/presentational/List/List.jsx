import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const List = ({ characters }) => {

  const charList = characters.map(char => {
    return (<li key={ char._id}>
      <Link  to={`/characters/${char._id}`}>{char.name}</Link>;
    </li>);
  });
  return (
    <ul>
      {charList}
    </ul>
  );
};

List.propTypes = {
  characters: PropTypes.array.isRequired
};

export default List;
