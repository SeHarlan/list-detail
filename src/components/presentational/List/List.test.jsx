import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('List component', () => {
  it('renders List', () => {
    const wrapper = shallow(<List characters={[{ _id: 1, name: 'aang' }]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
