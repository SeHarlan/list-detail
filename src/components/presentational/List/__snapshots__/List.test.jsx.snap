// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<ul>
  <li
    key="1"
  >
    <Link
      to="/characters/1"
    >
      aang
    </Link>
    ;
  </li>
</ul>
`;
